CenterForOpenScience/waterbutler
# [SVCS-576] Scrub keen payloads to make them valid

## Ticket
https://openscience.atlassian.net/browse/SVCS-576



## Purpose
Scrub '.' from keys in keen payloads. 



## Changes
'.' are now removed from keen payloads. If this created a conflict with another key, a (1) will be added to the end of it. Increments based on number of collisions
Added tests for new scrubbing function



## Side effects
Should be none



## QA Notes
To test this locally, you need to make a keen account at https://keen.io/ and then add your keys to
the settings file [here](https://github.com/CenterForOpenScience/waterbutler/blob/develop/waterbutler/settings.py#L165-L166) (the private keys)

Verify that your key is working by browsing a file or downloading a file on the osf. You should get some data to show up on your Keen.

Next, you need to send a bad payload. The best way to do this is to manually add code [here](https://github.com/CenterForOpenScience/waterbutler/blob/develop/waterbutler/core/remote_logging.py#L181) something like:
```
payload[test.....in....name] = 'dict value'
```

Once logged to keen, the value should appear as `testinname = 'dict value'`
Name incrementation can be tested like this:

```
payload['test....key] = 'some value'
payload[test.key] = 'other value'
```
once scrubbed, this will create a conflict. it will be solved by one of the keys turning into a new key

```
'testkey' : .... ,
'testkey (1)` value, ...
```

Unsure of best way to test this on staging, or how to send a dummy payload there.



## Deployment Notes
